.github/workflows/workflow.yml

```diff
@@ -0,0 +1,75 @@
+name: Main workflow
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          - windows-latest
+        # Please keep the list in sync with the minimal version of OCaml in
+        # dune-project, dune.opam.template and bootstrap.ml
+        ocaml-version:
+          - 4.10.0
+        skip_test:
+          - false
+        include:
+          - ocaml-version: 4.09.1
+            os: ubuntu-latest
+            skip_test: true
+          - ocaml-version: 4.08.1
+            os: ubuntu-latest
+            skip_test: true
+          - ocaml-version: 4.03.0
+            os: ubuntu-latest
+            skip_test: true
+          - ocaml-version: 4.02.3
+            os: ubuntu-latest
+            skip_test: true
+
+    # We don't run tests on all versions of the Windows environment and on
+    # 4.02.3 and 4.07.1 in other environments
+    env:
+      # Note that this environment variable is recognized as a string by GitHub
+      # Actions, so the following doesn't actually work:
+      #
+      # if: env.SKIP_TEST
+      #
```

Comment:
This should work, though, right: `if: ${{ env.SKIP_TEST != "false" }}`? The benefits are a slight reduction in runtime (since no container/VM is spun up) and also the clear UI (if GH Actions logs may at present be described as clear!) that it was skipped.

Reply:
Yes, this works. However, we have no use for it anymore. The `ci.ml` script takes care of this logic. Previously, we tried embedding this logic in the yaml but it was too ugly.

Reply:
Embedding the logic in the YAML is superior though - the UI makes it clear that the action was skipped.

Reply:
There's also conditional behavior for Windows vs. Linux So there are really 4 different configurations. If you want to move this logic back into yaml, perhaps you can give it a try in a separate PR?

Reply:
No, that bit's different - the Windows/Unix part is conditional on _how_ you build and test it... this is changing the matrix for _if_ you test it!

Reply:
Alright, I moved the conditional check on `SKIP_TEST` on the test action.

Reply:
That definitely looks nicer to me in the log: https://github.com/dra27/dune/runs/1007036469